core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -473,6 +473,16 @@ UNWIND(	.cantunwind)
 	bl	cpu_is_ready
 
 #if defined (CFG_BOOT_SECONDARY_REQUEST)
+spin:
+	/* spin until secondary boot request */
+	wfe
+	bl  get_core_pos
+	ldr r1, =spin_table
+	add r1, r1, r0, lsl #2	/* size of uint32_t */
+	ldr r0, [r1]
+	cmp	r0, #0
+	beq	spin
+
```

Comment:
i was thinking to something like:
```
	/* spin until secondary boot request */
	bl      get_core_pos
	lsl     r0, #2                /* size of uint32_t */
	ldr     r1, =spin_table

spin:
	wfe
	ldr     r2, [r1, r0]
	cmp     r2, #0
	beq     spin

	/* get ns entry from ns_entry_addrs array */
	dmb
 	ldr     r1, =ns_entry_addrs
	ldr     r0, [r1, r0]
```

Take care to the `dmb.` I think it is required to not use a value loaded from a predictive execution.